fs/inode/fs_files.c

```diff
@@ -105,9 +106,16 @@ static int files_extend(FAR struct filelist *list, size_t row)
     }
   while (++i < row);
 
+  flags = enter_critical_section();
```

Comment:
the sequence to hold the critical section and fl_lock is difference from fs_getfilep, the deadlock will happen on SMP machine.

Reply:
Done, I will remove file locks and use critical sections to protect file lists